test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/TransactionTests.cs

```diff
@@ -3,35 +3,279 @@
+            }
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenABundleWithForeignReferenceInResourceBody_WhenSubmittingATransaction_ThenReferenceShouldNotBeResolvedAndProcessShouldSucceed()
+        {
+            var requestBundle = Samples.GetJsonSample("Bundle-TransactionWithForeignReferenceInResourceBody");
+
+            var fhirResponse = await Client.PostBundleAsync(requestBundle.ToPoco<Bundle>());
+
+            Assert.NotNull(fhirResponse);
+            Assert.Equal(HttpStatusCode.OK, fhirResponse.StatusCode);
+
+            foreach (var entry in fhirResponse.Resource.Entry)
+            {
+                IEnumerable<ResourceReference> references = entry.Resource.GetAllChildren<ResourceReference>();
+                foreach (var reference in references)
+                {
+                    // Asserting the conditional reference value before resolution
+                    Assert.Equal("urn:uuid:4a089b8a-b0a0-46a9-92da-c8b653aa2e73", reference.Reference);
+                }
+            }
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenATransactionBundleReferencesInResourceBody_WhenSuccessfulExecution_ReferencesAreResolvedCorrectlyAsync()
+        {
+            var requestBundle = Samples.GetJsonSample("Bundle-TransactionWithValidBundleEntry");
+
+            FhirResponse<Bundle> fhirResponse = await Client.PostBundleAsync(requestBundle.ToPoco<Bundle>());
+            Assert.NotNull(fhirResponse);
+            Assert.Equal(HttpStatusCode.OK, fhirResponse.StatusCode);
+
+            var bundleWithConditionalReference = Samples.GetJsonSample("Bundle-TransactionWithReferenceInResourceBody");
+
+            var bundle = bundleWithConditionalReference.ToPoco<Bundle>();
+
+            FhirResponse<Bundle> fhirResponseForReferenceResolution = await Client.PostBundleAsync(bundle);
+
+            Assert.NotNull(fhirResponseForReferenceResolution);
+            Assert.Equal(HttpStatusCode.OK, fhirResponseForReferenceResolution.StatusCode);
+
+            foreach (var entry in fhirResponseForReferenceResolution.Resource.Entry)
+            {
+                IEnumerable<ResourceReference> references = entry.Resource.GetAllChildren<ResourceReference>();
+
+                foreach (var reference in references)
+                {
+                    // Asserting the conditional reference value before resolution
+                    Assert.True(reference.Reference.Contains("/", System.StringComparison.Ordinal));
+                }
+            }
+        }
+
+        private void ValidateResourceOutput(Bundle resource)
+        {
+            Assert.True("201".Equals(resource.Entry[0].Response.Status), "Create");
+            Assert.True("201".Equals(resource.Entry[1].Response.Status) || "200".Equals(resource.Entry[1].Response.Status), "Conditional Create");
```

Comment:
What is the case that these can be different? Ideally we would get the same response every time we ran the test.

Reply:
That's a good point. I reason i did this was i saw 200k and 201 created for conditional creates and updates. But that was before i implemented cleanup of resources before executing tests. Will update it.